### ISSUE TITLE
Enhancement: [no-floating-promises] Ignore `Promise.allSettled`

### ISSUE THREAD

**Issue by haines** [typescript-eslint/typescript-eslint#6304]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-floating-promises/

### Description

`Promise.allSettled` never rejects, so it doesn't make sense that this rule forces you to handle its rejection.

### Fail

```typescript
Promise.all(promises).then(console.log);
```

### Pass

```typescript
Promise.allSettled(promises).then(console.log);
```

### Additional Info

It's possible to work around the rule (assuming `ignoreVoid: true`) by writing `void Promise.allSettled(...)` instead.

**Comment by bradzacher** [comment]
> so it doesn't make sense that this rule forces you to handle its rejection

The rule isn't just about rejection - it's about resolution too.

`Promise.allSettled` may not throw, but it does execute asynchronously. Failing to await its result may result in out-of-order execution. Failing to handle it would likely be a bug in the vast majority of code.

The `void` operator workaround is intended for this exact circumstance of clearly documenting "there is a promise here that I am intentionally not awaiting".
